internal/controllers/reconciliation/controller.go

```diff
@@ -244,8 +278,8 @@ func (c *Controller) buildPatch(ctx context.Context, prev, resource *reconstitut
 	return patch, types.StrategicMergePatchType, err
 }
 
-func (c *Controller) getCurrent(ctx context.Context, resource *reconstitution.Resource, apiVersion string) (*unstructured.Unstructured, bool, error) {
-	if resource.HasBeenSeen() && !resource.Deleted() {
+func (c *Controller) getCurrent(ctx context.Context, resource *reconstitution.Resource, apiVersion string, disableCache bool) (*unstructured.Unstructured, bool, error) {
```

Comment:
Where is the connection between disable cache and !ready coming from?

Reply:
We can't use the metadata API to check the resource version when the resource isn't ready, since we wouldn't get the signal required to mark the resource as ready. There are some related comments up higher in the file, lmk if you think there's a more clear way to document this